src/ebmc/ranking_function.cpp

```diff
@@ -134,8 +135,11 @@ tvt is_ranking_function(
   const exprt &property,
   const exprt &ranking_function,
   const ebmc_solver_factoryt &solver_factory,
+  std::optional<trans_tracet> &witness_trace,
```

Comment:
Not a very big deal, but how about returning `std::pair<tvt, std::optional<trans_tracet>>` (or perhaps create a `struct ranking_function_resultt`)?